flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/PendingCheckpoint.java

```diff
@@ -384,31 +384,25 @@ public TaskAcknowledgeResult acknowledgeTask(
             long ackTimestamp = System.currentTimeMillis();
 
             long stateSize = 0L;
+            for (OperatorIDPair operatorID : operatorIDs) {
```

Comment:
Is this commit really needed? Conceptually it should only apply to rescaling.

Reply:
I was not sure. That's actually why I am going through a full review process for the backport ;) If you are saying it applies only for rescaling I'll drop the commit.